diff -Naur pam_tacplus.orig/Makefile.am pam_tacplus/Makefile.am
--- pam_tacplus.orig/Makefile.am	2018-10-19 15:31:02.612848223 +0200
+++ pam_tacplus/Makefile.am	2018-10-19 15:31:30.097547771 +0200
@@ -15,6 +15,7 @@
 tacc_CFLAGS = $(AM_CFLAGS) -I $(top_srcdir)/libtac/include
 tacc_LDFLAGS = -L.libs -ltac $(OPENSSL_LIBS)
 dist_man_MANS = tacc.1 tacplus_servers.5
+tacc_LDADD = libtac.la
 
 
 # This applies to all the libs, but it should only be for pam_tacplus.so
